packages/fast-components-react-base/src/dialog/dialog.tsx

```diff
@@ -140,6 +153,20 @@ class Dialog extends Foundation<DialogHandledProps, DialogUnhandledProps, {}> {
             this.props.onDismiss(event);
         }
     };
+
+    private handleDocumentFocus = (event: FocusEvent): void => {
+        if (isNil(this.rootElement.current)) {
+            return;
+        }
+        if (
+            this.props.visible &&
+            !(this.rootElement.current as HTMLElement).contains(
+                event.relatedTarget as HTMLElement
+            )
+        ) {
+            this.rootElement.current.focus();
```

Comment:
This would seem to suggest that if focus leaves the modal it will refocus on the modal, however that doesn't seem to support what aria specs have which is:

- Tab from last focusable element in the modal should focus on first focusable element,
- Shift + Tab on first focusable element should focus on last focusable element.

It seems like what we should do here is:
- Store first focusable item
- Store last focusable item
- Check during appropriate lifecycles in case a new focusable item has been rendered to see if there is a new first or last focusable item
- When the keydown event is Shift + Tab on the first item, focus on the last item
- When the keydown event is Tab on the last item, focus on the first item

You could probably use the current `handleWindowKeyDown` method.

Reply:
I thought about that and came to the conclusion that developers can manage their own focus loops in dialog ui.  When we refocus the dialog focus goes back to the last thing focused.  Why get involved in managing that if we don't have to?

Reply:
(there are existing implementations of these dialogs that already have focus loops)

Reply:
It seems like an issue may be useful here to actually track feedback and the problem a bit more. If the goal is to trap focus within modal dialogs, we should investigate ways it can be accomplished now (if at all) and then move towards a proposed fix/feature.

I'm also fine if we want to have the discussion here, but it may be difficult to track in this form.

Either way, I'm waiting on this PR as it's "live" and not a "draft" until we can land the overall problem, the current implementation, and then what's missing.

Reply:
Happy to move discussion wherever, the intent of the all caps "PROPOSED" tag was to indicate that I consider this a decision of consequence that needs discussion (and I tend towards having an implementation to discuss over).  Created an issue here - https://github.com/microsoft/fast-dna/issues/2329